src/UI/Component/Signal.php

```diff
@@ -12,4 +12,18 @@
  */
 interface Signal {
 
+	/**
+	 * Get the ID of this signal
+	 *
+	 * @return string
+	 */
+	public function getId();
```

Comment:
Please take a look at the following example: https://github.com/studer-raimann/ILIAS/blob/feature/5-3/ui-service-popovers-3/src/UI/examples/Popover/show_popover_with_dynamic_changing_content.php#L38

On this line, I'm passing the ID of the replace signal _explicitly_ as GET parameter to the async URL, thus I need the ID also as consumer. I agree that this sort of stuff should be handled internal by the framework, but I have not yet found a good solution. This example is more a "Proof of Concept", it works pretty well but is not yet abstracted perfectly by the framework. I'm looking forward to the discussion ;-)